openshift/topics/advanced_concepts.adoc

```diff
@@ -594,6 +594,54 @@ There are two methods for passing the {project_name} adapter configuration to th
 
 See xref:Example-EAP-Manual[Example Workflow: Manually Configure an Application to Use {project_name} Authentication, Using SAML Client] for an end-to-end example of the manual {project_name} client registration method using a SAML client.
 
+=== Using {project_name} vault with OpenShift secrets
+. Specify a directory for the vault using the *_SSO_VAULT_DIR_* environment variable. 
+You can introduce the *_SSO_VAULT_DIR_* environment variable directly in the environment in your deployment configuration. It can also be included in the template by addding the following snippets at the appropriate places in the template:
++
+[source,json,subs="attributes+,macros+"]
+----
+"parameters": [
+    ...
+    {
+        "displayName": "RH-SSO Vault Secret directory",
+        "description": "Path to the RH-SSO Vault directory.",
+        "name": "SSO_VAULT_DIR",
+        "value": "",
+        "required": false
+    }
+    ...
+]
+
+env: [
+    ...
+    {
+        "name": "SSO_VAULT_DIR",
+        "value": "${SSO_VAULT_DIR}"
+    }
+    ...
+]
+----
+
+. Create a secret in your OpenShift cluster:
++
+[source,bash,subs="attributes+,macros+"]
+----
+$ oc create secret generic rhsso-vault-secrets --from-literal=master_smtp__password=mySMTPPsswd
+----
+
+. Mount a volume to your deployment config using the `${SSO_VAULT_DIR}` as the path.
+For a deployment that is already running:
++
+[source,bash,subs="attributes+,macros+"]
+----
+oc set volume dc/sso --add --mount-path=${SSO_VAULT_DIR} --secret-name=rhsso-vault-secrets
+----
+
+. Now after a pod is created you can use a customized string within your {project_name} 
+configuration to refer to the secret. For example, for using `mySMTPPsswd` secret 
+created in this tutorial, you can use `${vault.smtp_password}` within `realm-name` 
+realm in the configuration of the smtp password and it will be replaced by `mySMTPPsswd` when used.
+
```

Comment:
@martin-kanis LGTM. 

Nitpicking thoug: Is the vault support / feature for the RH-SSO for OpenShift image somehow related with the Keycloak / RH-SSO server's SMTP (email) configuration? Or was that example chosen just by coincidence?

If the latter (the vault support in the image is not limited to / specific to the SMTP server password configuration in the image), I would rather prefer a more general example. Or if `${value.smtp_password}` is the preferred form / example to keep, I would highlight in the documention yet, the above SMTP configuration is just one example of (more) possible vault configuration, but it can be used for other use cases too.

Maybe something like `For example, to configure SMTP password for the {project_name} server using the vault feature, you can use `${vault.smtp_password]` ....`?

The rest LGTM. Thanks!